framework/src/executioners/Eigenvalue.C

```diff
@@ -61,40 +77,228 @@ Eigenvalue::validParams()
     paramError("normal_factor",
                "Cannot set scaling factor without defining normalization postprocessor.");
+
+  // _feproblem_solve calls FEProblemBase
+  _picard_solve.setInnerSolve(_feproblem_solve);
+  _time = _system_time;
+
+  if (isParamValid("normalization"))
+  {
+    auto normpp = getParam<PostprocessorName>("normalization");
+    if (isParamValid("normal_factor"))
+      _eigen_problem.setNormalization(normpp, getParam<Real>("normal_factor"));
+    else
+      _eigen_problem.setNormalization(normpp);
+  }
 }
 
 void
 Eigenvalue::init()
 {
-#if LIBMESH_HAVE_SLEPC
+  if (_app.isRecovering())
+  {
+    _console << "\nCannot recover eigenvaue solves!\nExiting...\n" << std::endl;
+    return;
+  }
+
+#ifdef LIBMESH_HAVE_SLEPC
   // Set a flag to nonlinear eigen system
   _eigen_problem.getNonlinearEigenSystem().precondMatrixIncludesEigenKernels(
       getParam<bool>("precond_matrix_includes_eigen"));
 #endif
-  Steady::init();
+
+  if (isParamValid("normalization"))
+  {
+    auto normpp = getParam<PostprocessorName>("normalization");
+    const auto & exec = _eigen_problem.getUserObject<UserObject>(normpp).getExecuteOnEnum();
+    if (!exec.contains(EXEC_LINEAR))
+      mooseError("Normalization postprocessor ", normpp, " requires execute_on = 'linear'");
+  }
+
+  // Does not allow time kernels
+  checkIntegrity();
+  // Some setup
+  _eigen_problem.execute(EXEC_PRE_MULTIAPP_SETUP);
+  _eigen_problem.initialSetup();
+
+#ifdef LIBMESH_HAVE_SLEPC
+
+  // Make sure all PETSc options are setup correctly
+  prepareSolverOptions();
+
+  // Let do an initial solve if a nonlinear eigen solver but not power is used.
+  // The initial solver is a Inverse Power, and it is used to compute a good initial
+  // guess for Newton
+  auto free_power_iterations = getParam<unsigned int>("free_power_iterations");
+  if (free_power_iterations && _eigen_problem.isNonlinearEigenvalueSolver() &&
+      _eigen_problem.solverParams()._eigen_solve_type != Moose::EST_NONLINEAR_POWER)
+  {
+    _eigen_problem.doFreePowerIteration(true);
+    // Set free power iterations
+    setFreeNonlinearPowerIterations(free_power_iterations);
+
+    // Provide vector of ones to solver
+    // "auto_initialization" is on by default and we init the vector values associated
+    // with eigen-variables as ones. If "auto_initialization" is turned off by users,
+    // it is up to users to provide an initial guess. If "auto_initialization" is off
+    // and users does not provide an initial guess, slepc will automatically generate
+    // a random vector as the initial guess. The motivation to offer this option is
+    // that we have to initialize  ONLY eigen variables in multiphysics simulation.
+    if (_eigen_problem.needInitializeEigenVector())
+      _eigen_problem.initEigenvector(1.0);
+
+    _console << " Free power iteration starts" << std::endl;
+
+    // Call solver
+    _eigen_problem.solve();
```

Comment:
I am getting `Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 1` in one of my test runs here. I suggest we clear the snes tolerances so that we do not get this.

Reply:
Good suggestion